src/NServiceBus.Core/Sagas/SagaMessage.cs

```diff
@@ -10,16 +12,22 @@ public class SagaMessage
         /// </summary>
         /// <param name="messageType">Type of the message.</param>
         /// <param name="isAllowedToStart"><code>true</code> if the message can start the saga, <code>false</code> otherwise.</param>
-        public SagaMessage(string messageType, bool isAllowedToStart)
+        public SagaMessage(Type messageType, bool isAllowedToStart)
         {
             MessageType = messageType;
+            MessageTypeName = messageType.FullName;
             IsAllowedToStartSaga = isAllowedToStart;
         }
 
         /// <summary>
         /// The type of the message.
         /// </summary>
-        public string MessageType { get; private set; }
+        public Type MessageType { get; private set; }
```

Comment:
I had some ideas that we should stop relying on CLR types to represent sagas (hence the string instead of Type). Not sure how valid that is, fine with YAGNI for now but curious if you actually need the type?